logs_sdk/lib/opentelemetry/sdk/logs/log_record.rb

```diff
@@ -139,10 +139,10 @@ def validate_attributes(attrs)
           # Future refactor opportunity: https://github.com/open-telemetry/opentelemetry-ruby/issues/1739
           attrs.keep_if do |k, v|
             if !Internal.valid_key?(k)
-              OpenTelemetry.handle_error(message: "invalid log record attribute key type #{k.class} on record: '#{body}'")
+              OpenTelemetry.logger.warn("Invalid log record attribute key type #{k.class} for key #{k.inspect} on record: '#{body}'. Attribute keys must be Strings. Dropping attribute.")
```

Comment:
This is going to produce very noisy log output. 

Do we log warnings in the SDK for traces?

Reply:
Great point. The Trace SDK does not use the logger to warn, instead, we handle the error like the current behavior in the logger. I wonder if updating the message for the `handle_error` call would be sufficient to let people know about the problem?

Here's the SDK logging code: https://github.com/open-telemetry/opentelemetry-ruby/blob/98c629edbc6445305bed65c4f4306431d683b474/sdk/lib/opentelemetry/sdk/internal.rb#L51-L63

Reply:
I think the main issue with the previous message was that the `key` was missing from the output. That made it more difficult to identify what attribute was the problem. I'm fine with keeping it as a `handle_error` call or using this PR to move it to a `logger.warn`.

I might lean more toward the `handle_error` approach so that people can define their own error handlers to control the output behavior.